src/test/ui/impl-trait/issues/issue-78722.stderr

```diff
@@ -15,7 +15,7 @@ LL | pub const fn from_generator<T>(gen: T) -> impl Future<Output = T::Return>
    |                                           ------------------------------- the found opaque type
    |
    = note: expected opaque type `impl Future<Output = u8>`
-              found opaque type `impl Future`
+              found opaque type `impl Future<Output = <[static generator@$DIR/issue-78722.rs:12:26: 12:31] as Generator<ResumeTy>>::Return>`
```

Comment:
I think though that we *should* special case `Generator` and call it something like `[async@$DIR/issue-78722.rs:12:20: 12:31]` so it looks like:

```
found opaque type `impl Future<Output = <[async@$DIR/issue-78722.rs:12:26: 12:31]>`
```

What do you think?

Reply:
Hmm.. Yeah, I could see us special-casing it. I wouldn't mind printing `[async@...]` for the whole `impl Future` type..

... unless there's a way of normalizing that `<_ as Generator>::Return` type in the code here. That would be the best.

Reply:
If you make `Generator::Return` a `lang_item` and query it here to unconditionally replace it with `_` or `[async result]`, what is the diff in the tests? We might want to go that route instead, depending on how much real code is affected. I suspect that this type isn't being shown much today because it most commonly appears only as part of `Future::Output`, so it wouldn't be a regression to hide it.